vllm/distributed/parallel_state.py

```diff
@@ -291,7 +288,33 @@ def all_reduce(self, input_: torch.Tensor) -> torch.Tensor:
             ipex.distributed.all_reduce(input_, group=self.device_group)
         else:
             torch.distributed.all_reduce(input_, group=self.device_group)
-        return input_
+
+    def out_of_place_ar(self, input_: torch.Tensor) -> torch.Tensor:
```

Comment:
where do you call this function?

Reply:
https://github.com/vllm-project/vllm/pull/7755/files#diff-4f562f5059f3e90ace1659583afc6fff378aa8db722d8c61995caef06df94cdaR14